include/thread.h

```diff
@@ -25,6 +27,9 @@ typedef struct thread {
   TAILQ_ENTRY(thread) td_zombieq; /* a link on zombie queue */
   char *td_name;
   tid_t td_tid;
+  /* Locks*/
+  mtx_t td_lock;
+  condvar_t td_waitcv;
```

Comment:
Please move these to the top of the structure and describe the purpose of `td_waitcv`.